langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/VertexAiImagenImageModel.java

```diff
@@ -0,0 +1,403 @@
+
+                PredictResponse predictResponse =
+                    withRetry(() -> client.predict(this.endpointName, instances, parameters), this.maxRetries);
+
+                List<Image> allImages = predictResponse.getPredictionsList().stream()
+                    .map(v -> {
+                        String bytesBase64Encoded = v.getStructValue()
+                            .getFieldsMap()
+                            .get("bytesBase64Encoded")
+                            .getStringValue();
+                        return Image.builder()
+                            .base64Data(bytesBase64Encoded)
+                            .url(persistAndGetURI(bytesBase64Encoded))
+                            .build();
+                    })
+                    .collect(Collectors.toList());
+
+                return Response.from(allImages);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private URI persistAndGetURI(String bytesBase64Encoded) {
+        if (this.withPersisting != null && this.withPersisting) {
+            try {
+                Path tempFile = Files.createTempFile(this.tempDirectory, "imagen-image-", ".png");
+                Files.write(tempFile, Base64.getDecoder().decode(bytesBase64Encoded));
+                return tempFile.toUri();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            return null;
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private String endpoint;
+        private String project;
+        private String location;
+        private String publisher;
+        private String modelName;
+        private Long seed;
+        private String language;
+        private Image image;
+        private Image mask;
+        private String negativePrompt;
+        private ImageStyle sampleImageStyle;
+        private Integer sampleImageSize;
+        private Integer maxRetries;
+        private Integer guidanceScale;
+        private Boolean withPersisting;
+        private Path persistTo;
+
+        public Builder endpoint(String endpoint) {
+            this.endpoint = endpoint;
+            return this;
+        }
+
+        public Builder project(String project) {
+            this.project = project;
+            return this;
+        }
+
+        public Builder location(String location) {
+            this.location = location;
+            return this;
+        }
+
+        public Builder publisher(String publisher) {
+            this.publisher = publisher;
+            return this;
+        }
+
+        public Builder modelName(String modelName) {
+            this.modelName = modelName;
+            return this;
+        }
+
+        public Builder seed(Long seed) {
+            this.seed = seed & 0xffffffffL;
```

Comment:
Do I understand it correctly that this masking ensures that provided `seed` is in a valid range? If yes, I would propose to do `ensureBetween(seed, MIN, MAX, "seed")` in the constructor instead, to fail fast if user provides wrong value instead of silently modifying it.

Reply:
The seed in an unsigned int, from 0 to 2^32, so I used a Long to make it fit, and the mask to keep it within bounds as you noticed. I'll go with the `ensureBetween()` method, it's a better idea than doing that transformation silently!

Reply:
I'll add a `Long` variant for that method, in addition to the existing `Integer` one (and `Double`).
(other option could be to just have a `Long` version)